balancer/balancer.go

```diff
@@ -45,54 +46,43 @@ type balanceEntity interface {
 }
 
 type peer struct {
-	hostname    string
-	entities    []balanceEntity
-	leaderCount int
-	rebalance   int
+	name     string
+	entities []balanceEntity
+	offset   int
 }
 
 // New returns a new instance of the Balancer
 func New(nc *nats.Conn, log api.Logger) (*Balancer, error) {
-	return &Balancer{
-		nc:  nc,
-		log: log,
-	}, nil
-}
+	mgr, err := jsm.New(nc)
+	if err != nil {
+		return nil, err
+	}
 
-func (b *Balancer) updateServersWithExclude(servers map[string]*peer, exclude string) (map[string]*peer, error) {
-	updated := map[string]*peer{}
-	var err error
+	apiLvl, err := mgr.MetaApiLevel(true)
+	if err != nil {
+		return nil, err
+	}
 
-	for _, s := range servers {
-		if s.hostname == exclude {
-			continue
-		}
-		for _, e := range s.entities {
-			updated, err = b.mapEntityToServers(e, updated)
-			if err != nil {
-				return updated, err
-			}
-		}
+	if apiLvl != 1 {
```

Comment:
check for `< 1` as 2.12 might be level 2 etc, each time we add a major API change it would be incremented